estate/models/est_estate_property_offer.py

```diff
@@ -0,0 +1,71 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import api, fields, models
+from datetime import datetime
+from odoo.exceptions import ValidationError, UserError
+
+
+class EstatePropertyOffer(models.Model):
+    _name = "est.estates.property.offer"
+    _description = "Estates Property Offer"
+    _order = "price desc"
+
+    price = fields.Float(default=0)
+    status = fields.Selection(
+        selection=[('accepted', 'Accepted'), ('refused', 'Refused')])
+    partner_id = fields.Many2one("res.partner", required=True)
+    property_id = fields.Many2one("est.estates.property", required=True, ondelete='cascade')
+    property_type_id = fields.Many2one(related='property_id.type_id')
+    validity = fields.Integer('Validity (days)', default=7)
+    date_deadline = fields.Date('Deadline', compute='_compute_deadline', inverse='_inverse_deadline', default=fields.Date.add(fields.Date.today(), days=7))
+
+    _sql_constraints = [
+        ('Check_price', 'CHECK(price > 0)',
+         'The offer price should be greater than 0!')
+    ]
+
+    @api.constrains('price')
+    def _check_price_against_expected_price(self):
+        for record in self:
+            expectedprice = self.property_id.expected_price / 100 * 90
+            if record.price < expectedprice:
+                raise ValidationError(('The selling price should be at least %s') % (expectedprice))
+
+    @api.model
+    def write(self, vals):
+        if vals.get('price'):
+            if self.search_count([('price', '>', vals['price']), ('property_id', '=', vals['property_id'])]) > 0:
+                raise UserError('You can create an offer with a price lower than an existing offer!')
+
+        return super().write(vals)
+
+    @api.model
+    def create(self, vals):
+        property = self.env['est.estates.property'].browse(vals['property_id'])
```

Comment:
here `property_id` will always be part of the vals during the `create`, but it's good practice to check nonetheless if it is in the vals dict first